src/Game/index.ts

```diff
@@ -488,10 +460,16 @@ export class Game {
 
   spawnWorker(team: Unit.TEAM, x: number, y: number, unitid?: string): Worker {
     const cell = this.map.getCell(x, y);
-    const unit = new Worker(x, y, team, this.configs, this.globalUnitIDCount++);
+    const unit = new Worker(
+      x,
+      y,
+      team,
+      this.configs,
+      this.globalUnitIDCount + 1
```

Comment:
@StoneT2000 Is this intentional? Look likes an off by 1 error.

Reply:
this is intentional. We increment this count later if we aren't given a id to use. If we are given an id to use, this is purely for resetting state and we expect the global is to be set elsewhere when resetting set (see toStateObj and the information stored there)